### PR TITLE
ENH: Allow to_sql to recognize single sql type #11886

### PR DESCRIPTION
**Pull request by RahulHP**
Follow-up in https://github.com/pydata/pandas/pull/13614

---

This solves #11886 
It checks whether the passed dtype variable is a dictionary.
If not, it creates a new dictionary with keys as the columns of the dataframe.
It then passes this dictionary to the pandasSQL_builder.

### CODE REVIEW

**Comment by RahulHP** [comment]
Could someone please tell me why this build failed? (https://travis-ci.org/pydata/pandas/jobs/132018332)
I can't find any error in the logs.

**Comment by jreback** [comment]
you have pep issues

https://travis-ci.org/pydata/pandas/jobs/132018332

**Comment by RahulHP** [comment]
``pandas/io/tests/test_sql.py:2050:80: E501 line too long (80 > 79 characters)
pandas/io/tests/test_sql.py:2051:53: E127 continuation line over-indented for visual indent
``
How should I fix these lines?
https://github.com/pydata/pandas/pull/13252/files#diff-052a926a9b2c33a84e27d51c8fad3466R2050

File: pandas/io/tests/test_sql.py [hunk 1]
```diff
@@ -2025,6 +2038,21 @@ def test_dtype(self):
...
+        if self.flavor == 'mysql':
+            raise nose.SkipTest('Not applicable to MySQL legacy')
+        self.drop_table('single_dtype_test')
+        cols = ['A', 'B']
+        data = [('a', 'b'),
+                ('c', 'd')]
+        df = DataFrame(data,columns=cols)
+        df.to_sql('single_dtype_test', self.conn, dtype='STRING')
+        self.assertEqual(self._get_sqlite_column_type('single_dtype_test', 'A'),
+                                                    'STRING')
```
Comment (jorisvandenbossche) [inline pandas/io/tests/test_sql.py]: You can do eg 

```
        self.assertEqual(
            self._get_sqlite_column_type('single_dtype_test', 'A'), 'STRING')
```

**Comment by jorisvandenbossche** [comment]
@RahulHP Is there a reason you closed this?

To clean up the unwanted commits here, you can do:

```
git fetch upstream/master
git rebase upstream/master
git push -f origin/master
```

**Comment by RahulHP** [comment]
I thought someone else would be better equipped to handle this. I have tried it but I seem to have trouble with the rebase related commands.

I'll try it again now.

**Comment by jorisvandenbossche** [comment]
@RahulHP You still have some commits that should not be in here. Normally, by doing exactly:

```
git fetch upstream/master
git rebase upstream/master
git push -f origin/master
```

it should be OK. If this does not work, can you show the output of those commands?

**Comment by jorisvandenbossche** [comment]
@RahulHP  Do you have time to update this?

**Comment by RahulHP** [comment]
No I don't, sorry for the trouble.

**Comment by jorisvandenbossche** [comment]
@RahulHP No problem, I will rebase this for you then, as it was a good enhancement!

**Comment by jorisvandenbossche** [comment]
Rebased in https://github.com/pydata/pandas/pull/13614

**Pull request closed without merging**